include/nanobind/eigen/dense.h

```diff
@@ -241,6 +249,107 @@ struct type_caster<Eigen::Ref<T, Options, StrideType>> {
     operator Ref() { return Ref(caster.operator Map()); }
 };
 
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = std::remove_reference_t<decltype(*std::declval<T>().outerIndexPtr())>;
+    using Index = typename T::Index;
+    static constexpr bool rowMajor = T::IsRowMajor;
+
+    using ScalarCaster = make_caster<tensor<Scalar>>;
```

Comment:
Your specification seems too general (we wouldn't want this to accept a matrix, non-contiguous representations, or GPU arrays). I would write
```cpp
using ScalarTensor = tensor<Scalar, shape<any>, c_contig, cpu>;
```

and then a few lines below:
```
make_caster<ScalarTensor> scalar_caster;
```